tests/swig.at

```diff
@@ -0,0 +1,69 @@
+# Souffle - A Datalog Compiler
+# Copyright (c) 2019, The Souffle Developers. All rights reserved
+# Licensed under the Universal Permissive License v 1.0 as shown at:
+# - https://opensource.org/licenses/UPL
+# - <souffle root>/licenses/SOUFFLE-UPL.txt
+
+m4_define([SWIG_LANGUAGES], [[java],
+  [python]])
+
+dnl Run driver program for given language to output csv files
+dnl $1 current testsuite directory
+dnl $2 language 
+m4_define([LANGUAGE_TEST],[
+  m4_if($2, java, [
+    javac *.java
+    java -Djava.library.path=$1 driver
+  ])
+  
+  m4_if($2, python, [
+    python3 driver.py
+  ])
+])
+
+dnl Positive testcase for invoking souffle with SWIG option given language and test name
+dnl $1 test name
+dnl $2 category
+dnl $3 language
+m4_define([TEST_EVAL_SWIG],[
+  m4_define([TESTNAME],[$1])
+  m4_define([CATEGORY],[$2])
+  m4_define([LANGUAGE],[$3])
+  m4_define([SCRIPTDIR],["$TESTS"/CATEGORY/LANGUAGE])
+  m4_define([TESTDIR],[SCRIPTDIR/TESTNAME])
+  m4_define([PROGRAM],[TESTDIR/TESTNAME.dl])
+  m4_define([FACTS],[TESTDIR/facts])
+  m4_define([EXPECTEDDIR],[expected])
+  m4_define([CURRENTDIR],[`pwd`])
+
+  for file in `ls TESTDIR`
+  do
+    cp -R "TESTDIR/$file" .
+  done

```

Comment:
Instead of copying the expected files and the java file to the current dir, you could just copy the files you need, e.g.,
```
  AS_IF([test LANGUAGE = "java"], [cp TESTDIR/*java .], [])
  AS_IF([test LANGUAGE = "python"], [cp TESTDIR/*py .], [])

```

Reply:
We needed to copy both the .dl and driver file as the driver file requires the .dl file to be in the same directory

Reply:
With the suggested change to main.cpp, the generated driver file would be in the current directory, rather than the dl directory, so there would be no need to copy it for testing.

Reply:
The driver.java and driver.py files are not generated they are already in the test directory and we copy these files along with the .dl and fact files

Reply:
Will fix, with an individual copy per file.

Reply:
Except for facts, which will use a wildcard (*.facts).